kirtonBCIlab/boccia-bci
# 84-user-can-calibrate-when-connected-to-ramp

Implementations
- Added `ISerialController` interface for `HardwareRamp` to implement.
- Mapped `HardwareRamp` methods `ConnectToSerialPort` and `DisconnectFromSerialPort` and mapped them to `BocciaModel`.
- Added functionality in `RampSetupPresenter` to connect, disconnect, and show errors in the serial connection.
- Added funcionality to mark when the motors are calibrated, drop button only enables when all motors are calibrated.

Testing
- To test the changes you will need an Arduino or some device to connect to the COM port.
- Open the device manager and navigate to "Ports".
- Check the number of COM port for the Arduino or device to connect.
- Uncomment the debug lines in `RampSetupPresenter` (95, 102, 114, 120).
- Select a COM port from the dropdown during gameplay and test the connection functionality.
- When an active serial connection, use the calibrate and recalibrate buttons
